aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/AuthHelper.kt

```diff
@@ -66,5 +69,27 @@ internal open class AuthHelper {
         fun getActiveUsername(username: String, alternateUsername: String?, userIDForSRP: String?): String {
             return alternateUsername ?: (userIDForSRP ?: username)
         }
+
+        /**
+         * This method is used to capture the username from the magic link.
+         * @param magicLinkCode the code from the magic link.
+         * @return the username from the magic link.
+         * @throws InvalidParameterException if the magiclink provided is malformed or invalid
+         * */
+        @Throws(InvalidParameterException::class)
+        fun getUsernameFromMagicLink(magicLinkCode: String): String {
+            return try {
+                val header = magicLinkCode.split(".")
+                if (header.size != 2) {
+                    throw InvalidParameterException("Magic Link is malformed.")
```

Comment:
What exception eventually gets logged to the customer? I'm not sure I'm seeing that in the feature test.

I would think it should be something like this.

AmplifyException("Magic Link is malformed", InvalidParameterException("Magic Link header size expected = 2, actual = x"